### PR TITLE
(wip) feat(infrastructure): setup development & demo-app build

### PR DESCRIPTION
**Pull request by ffriedl89**
Added a lot of tools and the necessary dependencies. 
I started at the development tasks for the build and worked my way through the dependencies.
There are still some tasks missing for the development.ts file, but i will add them.

It's not finished - i was not really sure what the demo-app should look like at the moment to be our MVP demo app. 
What do you think?

### CODE REVIEW

**Comment by thomaspink** [comment]
@ffriedl89 pls add a license header in all files that are based on the angular/material2 build.

Suggestion:
```ts
/**
 * @license
 * Based on the build of angular/material2 by Google Inc. governed by an 
 * MIT-style license that can be found in the LICENSE file at https://angular.io/license
 */
```

Opinions @ddprrt ?

File: tools/gulp/tasks/development.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+const {outputDir, packagesDir, projectDir} = buildConfig;
+
+/** Array of vendors that are required to serve the demo-app. */
+/** TODO: (ffr): check which vendor libs will be used in the demo-app */
+const appVendors = [
+  '@angular',
+  'systemjs',
+  'zone.js',
+  'rxjs',
+  // 'hammerjs',
```
Comment (thomaspink) [inline tools/gulp/tasks/development.ts]: remove line. we can add it later if needed

File: tools/gulp/tasks/development.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  'core-js',
+  // 'web-animations-js',
```
Comment (thomaspink) [inline tools/gulp/tasks/development.ts]: add line and npm install `web-animations-js`. we know already that we will need it

File: tools/gulp/tasks/development.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  // 'moment',
```
Comment (thomaspink) [inline tools/gulp/tasks/development.ts]: remove line for now

File: tools/package-tools/build-package.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+  /** Stores the secondary entry-points for this package if they haven't been computed already. */
+  private cacheSecondaryEntryPoints() {
+    if (!this._secondaryEntryPoints) {
+      this._secondaryEntryPointsByDepth = getSecondaryEntryPointsForPackage(this);
+      this._secondaryEntryPoints =
+        this._secondaryEntryPointsByDepth.reduce((list, p) => list.concat(p), []);
+    }
+  }
+}
```
Comment (thomaspink) [inline tools/package-tools/build-package.ts]: add new line at the end of the file (tslint)

File: tools/gulp/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    ],
+    "baseUrl": ".",
+    "paths": {
+      "ngx-groundhog-build-tools": ["../package-tools/index"]
+    }
+  },
+  "files": [
+    "gulpfile.ts"
+  ]
+}
```
Comment (thomaspink) [inline tools/gulp/tsconfig.json]: add new line at the end of the file

File: tools/package-tools/gulp/build-scss-task.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+// const gulpCleanCss = require('gulp-clean-css');
+
+/** Create a gulp task that builds SCSS files. */
+export function buildScssTask(outputDir: string, sourceDir: string, minifyOutput = false) {
+  return () => {
+    return src(join(sourceDir, '**/*.scss'))
+      .pipe(gulpSass().on('error', gulpSass.logError))
+      // .pipe(gulpIf(minifyOutput, gulpCleanCss()))
+      .pipe(dest(outputDir));
+  };
```
Comment (thomaspink) [inline tools/package-tools/gulp/build-scss-task.ts]: add new line at the end of the file (tslint)

File: tools/package-tools/gulp/sequence-task.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+/** Create a task that's a sequence of other tasks. */
+export function sequenceTask(...args: any[]) {
+  return (done: any) => {
+    gulpRunSequence(
+      ...args,
+      done
+    );
+  };
+}
```
Comment (thomaspink) [inline tools/package-tools/gulp/sequence-task.ts]: add new line at the end of the file (tslint)

File: tools/package-tools/gulp/watch-files.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+import {triggerLivereload} from './trigger-livereload';
+
+/**
+ * Function that watches a set of file globs and runs given Gulp tasks if a given file changes.
+ * By default the livereload server will be also called on file change.
+ */
+export function watchFiles(fileGlob: string | string[], tasks: string[], livereload = true,
+                            debounceDelay = 700) {
+  watch(fileGlob, {debounceDelay}, [...tasks, () => livereload && triggerLivereload()]);
+}
```
Comment (thomaspink) [inline tools/package-tools/gulp/watch-files.ts]: add new line at the end of the file (tslint)

File: tools/package-tools/secondary-entry-points.ts [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+      binary: 'findstr',
+      args: ['/r', `from.'@dynatrace/${packageName}/.*'`, `${searchDirectory}\\*.ts`]
+    };
+  } else {
+    return {
+      binary: 'grep',
+      args: ['-Eroh', '--include', '*.ts', `from '@dynatrace/${packageName}/.+';`, searchDirectory]
+    };
+  }
+}
```
Comment (thomaspink) [inline tools/package-tools/secondary-entry-points.ts]: add new line at the end of the file (tslint)

**Comment by ffriedl89** [comment]
license looks fine for me 👍 i will add it to the files
demo app is working now

**Comment by ffriedl89** [comment]
i removed the google headers in the demo app and added the license header in the tools files

**Pull request merged**